pykokkos/core/compiler.py

```diff
@@ -64,20 +64,30 @@ def compile_object(
         """
 
         metadata = module_setup.metadata
-
         hash: str = self.members_hash(metadata.path, metadata.name)
+        parser = self.get_parser(metadata.path)
+        entity: PyKokkosEntity = parser.get_entity(metadata.name)
+
+        
         if self.is_compiled(module_setup.output_dir):
             if hash not in self.members: # True if pre-compiled
+                if updated_types is not None:
```

Comment:
In what cases is `updated_types` expected to be `None`? Is it when the types are static?

Reply:
When there are no types to be inferred; the user already gave the annotations. I can either add a comment here to point this out or add it to the doc string, ... or both